internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -167,7 +167,19 @@ func (g *Generator) generate() (base *operatorsv1alpha1.ClusterServiceVersion, e
 	if g.FromVersion != "" {
 		base.Spec.Replaces = genutil.MakeCSVName(g.OperatorName, g.FromVersion)
 	}
-	base.Spec.RelatedImages = g.RelatedImages
+	relatedImageNamesMap := make(map[string]bool)
+	relatedImages := []operatorsv1alpha1.RelatedImage{}
+	for _, image := range append(base.Spec.RelatedImages, g.RelatedImages...) {
+		if _, found := relatedImageNamesMap[image.Name]; !found {
+			// append if name is not already in the list
+			relatedImageNamesMap[image.Name] = true
+			relatedImages = append(relatedImages, image)
+			continue
+		}
+		// if name is already in the list, error
+		return nil, fmt.Errorf("found more than one related image with name %s", image.Name)
+	}
```

Comment:
Sorry - I should have been more specific in my previous comment about de-duplicating.

Can we key this by the image field rather than the name? That's ultimately what matters. The relatedImage name is just informational and is not actually used by any of the mirroring tooling as far as I know.

We already have [some code](https://github.com/operator-framework/operator-sdk/blob/f62caa6579d14883155b91e241a98272d7a9c161/internal/cmd/operator-sdk/generate/internal/relatedimages.go#L125-L142) that does this for the `g.RelatedImages`. I wonder if we can combine `base.Spec.RelatedImages` earlier in the process such that that code picks them up and uses the same de-dup and warn logic?

In the case where there are multiple images with different names, I think @ryantking

Reply:
cc @ryantking

Reply:
As far as I can tell, the warning only exists for same ref but different name which I guess is all we care about.

Reply:
resolved via https://github.com/operator-framework/operator-sdk/commit/49622d5c939e6e22c5d54c3878697db811473116